src/zarr/core/array.py

```diff
@@ -4148,6 +4109,22 @@ def _parse_chunk_encoding_v3(
     else:
         out_array_bytes = _parse_array_bytes_codec(serializer)
 
+    if compressors is None:
+        out_bytes_bytes = ()
+
+    elif compressors == "auto":
+        out_bytes_bytes = default_bytes_bytes
+
+    else:
+        if isinstance(compressors, dict | Codec):
+            maybe_bytes_bytes = (compressors,)
+        elif compressors is None:
```

Comment:
Is it possible for this to be true? don't we already check if `compressors` is `None` on L4112?

Reply:
No, that is not possible. Fixed.